src/main/java/org/terracotta/statistics/AbstractOperationStatistic.java

```diff
@@ -63,9 +65,22 @@ public long sum() {
   public void begin() {
     if (!derivedStatistics.isEmpty()) {
       long time = Time.time();
+      operationStartTime.set(time);
       for (ChainedOperationObserver<? super T> observer : derivedStatistics) {
         observer.begin(time);
       }
     }
   }
+
+  @Override
+  public void end(T result) {
+    if (!derivedStatistics.isEmpty()) {
+      long time = Time.time();
+      long latency = Math.max(0, time - operationStartTime.get());
```

Comment:
Why would it be negative?

Reply:
`System.nanoTime()` can return a negative if the origin taken is in the future. Though, I should not use `Math.max()`. but `Math.asb()` instead.

Reply:
Why `abs`? That would reverse the latency. Why would the origin be in the future?

Reply:
from javadoc:

 _The value returned represents nanoseconds since some fixed but arbitrary origin time (perhaps in the future, so values may be negative). The same origin is used by all invocations of this method in an instance of a Java virtual machine; other virtual machine instances are likely to use a different origin._

Bt you are right I do not even need `abs` because if `System.nanoTime()` is negative, at time T0, whether it is negative or it has become positive at time T1, as long as we have T0 < T1 (consecutive calls), then T1 - T0 will always be positive.